org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/FindWindow.java

```diff
@@ -2010,8 +1998,8 @@ public void onSelectedQueryChanged() {
     	}
 		else {
 			MUserQuery uq = userQueries[index-1];
-			btnSave.setDisabled(!uq.userCanSave());
-			btnShare.setDisabled(!uq.userCanShare());
+			btnSave.setDisabled(!uq.userCanSave() || uq.getCode().startsWith("@SQL="));
+			btnShare.setDisabled(!uq.userCanShare() || uq.getCode().startsWith("@SQL="));
```

Comment:
isn't it cleaner to change userCanSave and userCanShare to return false if uq.getCode().startsWith("@SQL=") ?

Reply:
> isn't it cleaner to change userCanSave and userCanShare to return false if `uq.getCode().startsWith("@SQL=") `?

Thanks @hengsin, done with commit https://github.com/idempiere/idempiere/pull/1913/commits/f3378c563048a62504ed5fe17a588c5c1b9c0a1f